motr/obj.c

```diff
@@ -563,25 +578,30 @@ static int obj_namei_op_init(struct m0_entity *entity,
  */
 static int obj_op_obj_init(struct m0_op_obj *oo)
 {
-	int                     rc;
-	struct m0_locality     *locality;
-	struct m0_pool_version *pv;
-	struct m0_obj          *obj;
-	struct m0_client       *cinst;
+	int                      rc;
+	struct m0_locality      *locality;
+	struct m0_pool_version  *pv;
+	struct m0_obj           *obj;
+	struct m0_client        *cinst;
+	struct m0_layout_domain *ldom;
 
 	M0_ENTRY();
 	M0_PRE(oo != NULL);
 	M0_PRE(M0_IN(OP_OBJ2CODE(oo), (M0_EO_CREATE,
 				       M0_EO_DELETE,
 				       M0_EO_OPEN)));
 
-	/** Get the object's pool version. */
+	/** Get the object's pool version and optimal layout id. */
 	obj = m0__obj_entity(oo->oo_oc.oc_op.op_entity);
+	cinst = m0__obj_instance(obj);
```

Comment:
If this cinst is only needed for if (OP_OBJ2CODE(oo) == M0_EO_CREATE), you can reduce the scope of this variable and move this part of the code to 581

Reply:
cinst is used in 
`else if (OP_OBJ2CODE(oo) == M0_EO_OPEN) {`
as well